### ISSUE TITLE
Can't download local CA certificate anymore

### ISSUE THREAD

**Issue by tituspijean** [YunoHost/issues#1538]
Yunohost v3.7.0.8

One cannot download the local CA certificate from the webadmin. It leads to a 403 error:

![image](https://user-images.githubusercontent.com/8769166/77833661-75b69600-713f-11ea-9afe-cce552e06bd4.png)

**Comment by alexAubin** [comment]
Discussed on the chat or forum with another user, personally I would just vote to drop that feature, it dates back from the days before Let's Encrypt, but now it just make way less sense to download your own CA cert ...

**Comment by alexAubin** [comment]
Since somebody else reports the issue on the forum, I'm really curious if you have any use case where you need to download the CA

For now the feature is dropped in 3.8...

**Comment by tituspijean** [comment]
In my case, I have a YNH server on a Rasperry Pi, on a network I do not trust (I use Zerotier to mitigate that) . I also have no access to the router, so no port forwarding is possible.

Hence, I cannot use Let's Encrypt certificates at all, so I rely on the server's own CA. I then install it on all of my devices, so the browsers and apps stop complaining about insecure connections due to self-signed certificates.

I rest my case, your honor. 😉

**Comment by alexAubin** [comment]
Meh okay ... but that remains a super-specific case and you can still download the cert using SCP ... granted it's clearly more technical and complex than clicking a button ...

**Comment by tituspijean** [comment]
I totally agree my case is quite specific, or even unique. It seems there is (at least two?) other users who needed the file. Let's try to gather their need for this file? 

1. Question asked [here](https://forum.yunohost.org/t/yunohost-3-7-release-sortie-de-yunohost-3-7/10648/49?u=tituspijean)
2. ? Cannot find the other mention of the issue.

**Comment by SilverViper** [comment]
I download/save the certificates from within the browser for my test deployments.

**Comment by alexAubin** [comment]
@SilverViper : you mean otherwise the browser complains about the self-signed cert and doesn't let you add an exception ? Or for increased security in case somebody MitM in your network ? :s 

If that's about the browser, imho it's a bug that should be adressed somehow ... I thought we adressed it multiple times but it comes back again and it's never clear to me what makes the browser to refuse adding an exception...

**Comment by SilverViper** [comment]
It depends per browser/application/os how to handle exceptions. Most browsers let you view and save the certificate chain. When you add the root CA certificate to the Windows Certificate Store, in the Trusted Root Certificates folder, most applications will trust the server. It trusts the Let's Encrypt certificates in the same fashion, their root is in the WCS. Not every application uses the WCS; Java, Firefox, and iirc Adobe stuff don't do by default, they have their own certificate stores. So you will need to manually add it there as well.

HSTS makes some browsers even more stubborn on loading sites with certificate issues and remove the 'proceed anyway' link.

Once you have the self signed root CA in your certificate store, any MitM attack will be impossible as they don't have the private key used to generate the certificate unless they have direct access to the server to steal it. But what's the point of a MitM attack if you already control the server? Alternatively, they can generate their own certificates without it, but it won't be signed with the certificate that's in your store and you will get alerted for an untrusted certificate/chain.

Then there's a chicken and egg problem: you download the certificate over a connection you don't trust to make future connections trusted... How can you tell it hasn't been altered? :)

**Comment by tituspijean** [comment]
Would it be worth reconsidering the matter, considering the .local domains management that is coming in v4.3?

**Comment by alexAubin** [comment]
@tituspijean : meh do you still need this ? I still think this looks like a super edgy case and can be solved by just `scp`-ing the corresponding file x_x

**Comment by tituspijean** [comment]
I can indeed be solved with `scp`, but it is not great UX. 😇

**Comment by zamentur** [comment]
May be we need a documentation on this, instead of a feature.

An other way is to suggest the download of the CA only if you have a .local domain installed.

**Comment by tituspijean** [comment]
Considering the intranet use cases, I would just add a link to the CA file any time a domain does not have a Let's Encrypt certificate.
